src/pyfmi/fmi_base.pyx

```diff
@@ -438,6 +430,9 @@ cdef class ModelBase:
                 self._log_stream.seek(0)
             except AttributeError:
                 raise FMUException("In order to extract the XML-log from a stream, it needs to support 'seek'")
+        else:
+            if self.file_object:
```

Comment:
This ensures the `.txt` log is fully written out before trying to convert to `.xml`. There were some rare edge cases where this caused invalid XML.